homeassistant/components/crownstone/config_flow.py

```diff
@@ -0,0 +1,296 @@
+            elif auth_error.type == "LOGIN_FAILED_EMAIL_NOT_VERIFIED":
+                errors["base"] = "account_not_verified"
+        except CrownstoneUnknownError:
+            errors["base"] = "unknown_error"
+
+        # show form again, with the errors
+        if errors:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema(
+                    {vol.Required(CONF_EMAIL): str, vol.Required(CONF_PASSWORD): str}
+                ),
+                errors=errors,
+            )
+
+        await self.async_set_unique_id(self.cloud.cloud_data.user_id)
+        self._abort_if_unique_id_configured()
+
+        self.login_info = user_input
+        return await self.async_step_usb_config()
+
+    async def async_step_usb_config(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Set up a Crownstone USB dongle."""
+        list_of_ports = await self.hass.async_add_executor_job(
+            serial.tools.list_ports.comports
+        )
+        ports_as_string = list_ports_as_str(list_of_ports)
+
+        if user_input is not None:
+            selection = user_input[CONF_USB_PATH]
+
+            if selection == DONT_USE_USB:
+                return self.async_create_new_entry()
+            if selection == MANUAL_PATH:
+                return await self.async_step_usb_manual_config()
+            if selection != REFRESH_LIST:
+                selected_port: ListPortInfo = list_of_ports[
+                    (ports_as_string.index(selection) - 1)
+                ]
+                self.usb_path = await self.hass.async_add_executor_job(
+                    usb.get_serial_by_id, selected_port.device
+                )
+                return await self.async_step_usb_sphere_config()
+
+        return self.async_show_form(
+            step_id="usb_config",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_USB_PATH): vol.In(ports_as_string)}
+            ),
+        )
+
+    async def async_step_usb_manual_config(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manually enter Crownstone USB dongle path."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="usb_manual_config",
+                data_schema=vol.Schema({vol.Required(CONF_USB_MANUAL_PATH): str}),
+            )
+
+        self.usb_path = user_input[CONF_USB_MANUAL_PATH]
+        return await self.async_step_usb_sphere_config()
+
+    async def async_step_usb_sphere_config(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Select a Crownstone sphere that the USB operates in."""
+        spheres = {sphere.name: sphere.cloud_id for sphere in self.cloud.cloud_data}
+        # no need to select if there's only 1 option
+        if len(spheres) == 1:
+            user_input = {CONF_USB_SPHERE: next(iter(spheres.keys()))}
```

Comment:
Can you avoid replacing `user_input` here? When doing a security review it makes it harder to tell which data is coming from the user and which data is trusted.